packages/abi-coder/src/function-fragment.ts

```diff
@@ -215,7 +165,7 @@ export class FunctionFragment<
     if (outputAbiType.type === '()') return [undefined, 0];
 
     const bytes = arrayify(data);
-    const coder = AbiCoder.getCoder(this.jsonAbi, this.jsonFn.output);
+    const coder = AbiCoder.getCoder(new ResolvedAbiType(this.jsonAbi, this.jsonFn.output));
```

Comment:
Hmm, you're right, it was cleaner. However, the `ResolvedAbiType` is supposed to be a self-sufficient representation of a type where all of its generics have been resolved, no matter how deeply nested they are. Instantiating it _one time_ in the `getCoder` method as it currently is wouldn't be possible as this method is used recursively. 

There are two options:
1. Use your proposal, but create a private `getCoderImpl` method which takes a `ResolvedAbiType`, or
2. Overload the signatures of `getCoder` to `getCoder(abi: JsonAbi, argument: JsonAbiArgument)` and `getCoder(resolvedAbiType: ResolvedAbiType)`, and then have the implementation check if the first argument is an `instanceof ResolvedAbiType` and, if it's not, then create it because we're then sure that the `getCoder(abi, argument)` overload was used.

I've initially implemented the second option, but after some thought I realized that it adds more cognitive load than the first, so I've chosen the first option.